### PR TITLE
crtools: improve error handling on signal setting

### PR DESCRIPTION
**Pull request by ivalery111**
Signed-off-by: Valery Ivanov <ivalery111@gmail.com>

### CODE REVIEW

File: criu/crtools.c [hunk 1]
```diff
@@ -189,7 +189,8 @@ int main(int argc, char *argv[], char *envp[])
 	 * 2) Transmitting data to the image streamer
 	 * 3) Emitting logs (potentially to a pipe).
 	 */
-	signal(SIGPIPE, SIG_IGN);
+	if (signal(SIGPIPE, SIG_IGN) == SIG_ERR)
+		pr_warn("Failed to set a SIGPIPE signal ignore");
```
Comment (rst0git) [inline criu/crtools.c:193]: @ivalery111 could you please add `\n` at the end of the message?

Comment (ivalery111) [reply criu/crtools.c]: Yes, of course. My bad.

Comment (avagin) [reply criu/crtools.c]: ... : %s\n", strerror(errno)

Comment (avagin) [reply criu/crtools.c]: or you can use pr_perror in this case. And I would add "return 1" if it fails.

Comment (ivalery111) [reply criu/crtools.c]: Done, thanks.

Comment (rst0git) [reply criu/crtools.c]: In contrast with `pr_warn`, a new line character is appended to the end of the message with `pr_perror`.

```shell
$ make lint
...
# Do not append \n to pr_perror or fail
! git --no-pager grep -E '^\s*\<(pr_perror|fail)\>.*\\n"'
criu/crtools.c:		pr_perror("Failed to set a SIGPIPE signal ignore.\n");
make: *** [Makefile:414: lint] Error 1
```

Comment (ivalery111) [reply criu/crtools.c]: I didn't know about `make lint` option. Thank you.

**Review by rst0git — APPROVED** [review]

**Pull request merged**